### PR TITLE
Add JS summary handlers and some tests

### PR DESCRIPTION
**Pull request by na--**
Sorry the code is kind of :vomiting_face: I was restricted by the need to support k6's `base` compatibility mode, and I closely followed the Go code in k6 and Go's stdlib in the manual transpilation, so that the results of the first version maximally resemble the current k6 text summary... :disappointed:  We can change these things in the future, and we should definitely add more tests and probably refactor the code to be less of a pile of functions, but this should do for an alpha version, I think.

This closes https://github.com/loadimpact/k6/issues/1120, since it contains a very basic version of a JUnit-generating function.

### CODE REVIEW

**Comment by na--** [comment]
Erm despite the green checkmark, the tests seem to have been failing with a script error for a long time... See https://app.circleci.com/pipelines/github/loadimpact/jslib.k6.io/101/workflows/83419271-12e2-474f-a49b-aa395a8f5a62/jobs/224:
```
ERRO[0008] TypeError: Value is not an object: undefined
	at file:///home/circleci/jslib.k6.io/tests/testSuite.js:25:22(4)  executor=shared-iterations scenario=default source=stacktrace

running (00m00.1s), 0/1 VUs, 1 complete and 0 interrupted iterations
default ✓ [======================================] 1 VUs  00m00.0s/10m0s  1/1 shared iters

    data_received........: 0 B 0 B/s
    data_sent............: 0 B 0 B/s
    iteration_duration...: avg=42.55µs min=42.55µs med=42.55µs max=42.55µs p(90)=42.55µs p(95)=42.55µs
    iterations...........: 1   15.751111/s
```

No checks whatsoever, but passing tests... https://github.com/loadimpact/k6/pull/1769 will make this easier, but for now I'll add a `try / catch` block to fail the tests on script errors. And then fix the issue tomorrow.

**Comment by na--** [comment]
Here is a test script I use to compare the new JS summary and the old Go-based summary: https://gist.github.com/na--/d68652c21a83821d0542bd52850eae28

I basically compare `stdout-old.txt` and `stdout-new.txt` - there are some differences in the color positioning due to newlines, but they shouldn't matter. Instead of fancy tests for `textSummary`, I can make one that uses a snapshot of `raw-data.json` from that file and generates the summary from it, comparing it to the `stdout-new.txt` contents? After carefully verifying the summary by eye, that should be good enough, or at least better than nothing?

**Comment by legander** [comment]
Holy moly lots of GO code! Do you want me to review the code or just verifying the output?
Do you wanna release it as 0.0.1-alpha or is 0.0.1 enough?
Thanks for fixing the test case running! 🙏 
The formatting gotta stay `prettier` though, so it is going to get formatted to 2 spaces, sorry about that.

**Comment by na--** [comment]
> Holy moly lots of GO code! Do you want me to review the code or just verifying the output?

Yeah... Sorry for the Go-style of code, but as I said, I decided to almost literally translate what currently exists in k6 as the first step. Once we have that and some more tests, we can refactor it and remove superfluous things (`strWidth` probably needs to go or be significantly changed). So, basically, if you ignore the unidiomatic JS, can you take a quick look and see if anything seems wrong. I am especially concerned about the functions that work with numbers, since numbers are a bit finicky in JS, so basically `humanizeValue()` and all of its sub-functions. I've added tests for them, but I am pretty sure I have missed a few corner cases.

> Do you wanna release it as 0.0.1-alpha or is 0.0.1 enough?

0.0.1 is enough, I think, the code is at least usable...

> The formatting gotta stay `prettier` though, so it is going to get formatted to 2 spaces, sorry about that.

Ah, sorry, I missed that, will fix it after lunch

**Review by MStoykov — APPROVED** [review]
LGTM

**Comment by na--** [comment]
@legander, I fixed the formatter back to Prettier. I didn't have that VSCode module installed, so I think I just clicked `Yes` when it complained and suggested to change it to something else... And I don't really care about the indentation, though the lack of semicolons is giving me anxiety :sweat_smile:

**Review by imiric — COMMENTED** [review]
Mostly LGTM, just some nitpicks. Agree that having tests with the actual rendered summary would be good, as well as refactoring it to use "classes", etc., but we can address that later.

I'll give it a test later today.

File: lib/k6-summary/0.0.1/index.js [hunk 1]
```diff
@@ -0,0 +1,409 @@
...
+            return [
+                humanizeValue(metric.values.value, metric, timeUnit),
+                "min=" + humanizeValue(metric.values.min, metric, timeUnit),
+                "max=" + humanizeValue(metric.values.max, metric, timeUnit),
+            ];
+        case 'rate':
+            return [
+                humanizeValue(metric.values.rate, metric, timeUnit),
+                "✓ " + metric.values.passes,
+                "✗ " + metric.values.fails,
```
[the remark this answers is not in the document]
Comment (na--) [reply lib/k6-summary/0.0.1/index.js]: resolved by https://github.com/loadimpact/jslib.k6.io/pull/20/commits/79a112d696ff22ff25216f4f52e7f437bf7e3f52

File: lib/k6-summary/0.0.1/index.js [hunk 2]
```diff
@@ -0,0 +1,409 @@
...
+    result = (rem % 60) + 'm' + result;
+    rem = Math.trunc(rem / 60);
+    if (rem < 1) { // less than an hour
+        return result;
+    }
+    return rem + 'h' + result;
+}
+
+function humanizeDuration(dur, timeUnit) {
+    if (timeUnit != '' && unitMap.hasOwnProperty(timeUnit)) {
```
[the remark this answers is not in the document]
Comment (na--) [reply lib/k6-summary/0.0.1/index.js]: resolved by https://github.com/loadimpact/jslib.k6.io/pull/20/commits/79a112d696ff22ff25216f4f52e7f437bf7e3f52

File: lib/k6-summary/0.0.1/index.js [hunk 3]
```diff
@@ -0,0 +1,409 @@
...
+
+var groupPrefix = '█';
+var detailsPrefix = '↳';
+var succMark = '✓';
+var failMark = '✗';
+var defaultOptions = {
+    indent: ' ',
+    enableColors: true,
+    summaryTimeUnit: undefined,
+    summaryTrendStats: undefined,
```
[the remark this answers is not in the document]
Comment (na--) [reply lib/k6-summary/0.0.1/index.js]: resolved by https://github.com/loadimpact/jslib.k6.io/pull/20/commits/79a112d696ff22ff25216f4f52e7f437bf7e3f52

**Comment by na--** [comment]
@imiric I am currently adding a few simple tests that use `raw-data.json` from a real test and compare, will hopefully post them in the next hour. Not ideal, but better than nothing and would make the refactoring easier.

**Comment by legander** [comment]
Didn't get a chance yet to review, will hopefully be able to later today.

**Comment by na--** [comment]
@legander, thanks! I'm going to merge this in a bit, since we link to it in the v0.30.0 release notes, but if you find any major bugs in your review we can just update the `0.0.1` version, since we're not changing the API. Or release v0.0.2 and update the release notes.

**Comment by legander** [comment]
Aight! Don't forget to add it to supported.json and run `yarn run generate-homepage` to update the index page if you want it to show up under "Available libs" @na--

**Comment by na--** [comment]
Didn't know I had to do that :sweat_smile: Thanks! :bow:
